zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java

```diff
@@ -39,16 +40,20 @@
 final class ThrottledCall<V> extends Call.Base<V> {
   final ExecutorService executor;
   final Limiter<Void> limiter;
+  final Predicate<Throwable> isOverCapacity;
   final Call<V> delegate;
 
-  ThrottledCall(ExecutorService executor, Limiter<Void> limiter, Call<V> delegate) {
+  ThrottledCall(ExecutorService executor, Limiter<Void> limiter,
+    Predicate<Throwable> isOverCapacity, Call<V> delegate) {
     this.executor = executor;
     this.limiter = limiter;
+    this.isOverCapacity = isOverCapacity;
     this.delegate = delegate;
   }
 
   @Override protected V doExecute() throws IOException {
-    Listener limitListener = limiter.acquire(null).orElseThrow(RejectedExecutionException::new);
+    Listener limitListener = limiter.acquire(null)
+      .orElseThrow(RejectedExecutionException::new); // TODO: make an exception message
```

Comment:
FWIW, this is more of a safety check than anything.  The Limiter I implemented should never cause this exception to be thrown.

Reply:
if you recall the reason why it is unreachable mention. otherwise no biggie thx

Reply:
https://github.com/openzipkin/zipkin/blob/31338b52714c2dbb989b07beaa58cf7957d12e8c/zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledStorageComponent.java#L188

That is why :)